pkg/operator/sync.go

```diff
@@ -126,6 +144,70 @@ func (optr *Operator) syncTerminationHandler(config *OperatorConfig) error {
 	return nil
 }
 
+func (optr *Operator) syncWebhookConfiguration(config *OperatorConfig) error {
+	if err := optr.syncValidatingWebhook(
+		optr.kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations(),
+		mapiv1.MachineValidatingWebhook(config.TargetNamespace),
+		mapiv1.MachineSetValidatingWebhook(config.TargetNamespace),
+	); err != nil {
+		return err
+	}
+
+	return optr.syncMutatingWebhook(
+		optr.kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations(),
+		mapiv1.MachineMutatingWebhook(config.TargetNamespace),
+		mapiv1.MacheSetMutatingWebhook(config.TargetNamespace),
+	)
+}
+
+func (optr *Operator) syncValidatingWebhook(client admissionclientv1.ValidatingWebhookConfigurationInterface, hooks ...admissionregistrationv1.ValidatingWebhook) error {
+	hook := mapiv1.NewValidatingWebhookConfiguration(hooks...)
+	validHook, err := optr.validatingWebhookLister.Get(hook.Name)
+	if apierrors.IsNotFound(err) {
+		_, err = client.Create(context.Background(), hook, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return err
+	}
+	if !equality.Semantic.DeepDerivative(hook.Webhooks, validHook.Webhooks) {
+		updateHook := validHook.DeepCopy()
+		updateHook.Annotations = hook.Annotations
+		updateHook.Webhooks = hook.Webhooks
+		_, err = client.Update(context.Background(), updateHook, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (optr *Operator) syncMutatingWebhook(client admissionclientv1.MutatingWebhookConfigurationInterface, hooks ...admissionregistrationv1.MutatingWebhook) error {
+	hook := mapiv1.NewMutatingWebhookConfiguration(hooks...)
```

Comment:
why does NewMutatingWebhookConfiguration needs `...admissionregistrationv1.MutatingWebhook` in the signature?. IMO these abstractions add no value and result in unnecessary complexity. We know ahead of time all we need and we can be explicit about it. I think all we need is just two funcs with no parameters:
`NewMutatingWebhookConfiguration`
`NewValidatingWebhookConfiguration

Reply:
naming hook vs validHook seems confusing. Can we may be use expected vs current/got?